dlib/image_loader/webp_loader.cpp

```diff
@@ -0,0 +1,130 @@
+
+namespace dlib
+{
+
+    static std::vector<unsigned char> load_contents(const std::string &filename)
+    {
+        std::ifstream stream(filename);
+        stream.exceptions(std::ifstream::failbit | std::ifstream::badbit | std::ifstream::eofbit);
+        stream.seekg(0, std::ios_base::end);
+        std::vector<unsigned char> buffer(stream.tellg());
+        stream.seekg(0);
+        stream.read(reinterpret_cast<char*>(buffer.data()), buffer.size());
+        return buffer;
+    }
+
+// ----------------------------------------------------------------------------------------
+
+    webp_loader::
+    webp_loader( const char* filename ) : height_( 0 ), width_( 0 )
+    {
+        data = load_contents(filename);
+        get_info();
+    }
+
+// ----------------------------------------------------------------------------------------
+
+    webp_loader::
+    webp_loader( const std::string& filename ) : height_( 0 ), width_( 0 )
+    {
+        data = load_contents(filename);
+        get_info();
+    }
+
+// ----------------------------------------------------------------------------------------
+
+    webp_loader::
+    webp_loader( const dlib::file& f ) : height_( 0 ), width_( 0 )
+    {
+        data = load_contents(f.full_name());
+        get_info();
+    }
+
+// ----------------------------------------------------------------------------------------
+
+    webp_loader::
+    webp_loader( const unsigned char* imgbuffer, size_t imgbuffersize ) : height_( 0 ), width_( 0 )
+    {
+        data.resize(imgbuffersize);
+        memcpy(data.data(), imgbuffer, imgbuffersize);
+        get_info();
+    }
+
+// ----------------------------------------------------------------------------------------
+
+    void webp_loader::get_info()
+    {
+        if (!WebPGetInfo( data.data(), data.size(), &width_, &height_ )) {
+          throw image_load_error("webp_loader: Invalid header");
+        }
+    }
+
+// ----------------------------------------------------------------------------------------
+
+    void webp_loader::read_argb( unsigned char *out, const size_t out_size, const int out_stride ) const
+    {
+        if (!WebPDecodeARGBInto(data.data(), data.size(), out, out_size, out_stride)) {
+          throw image_load_error("webp_loader: decoding failed");
+        }
+    }
+
+// ----------------------------------------------------------------------------------------
+
+    void webp_loader::read_rgba( unsigned char *out, const size_t out_size, const int out_stride ) const
+    {
+        if (!WebPDecodeRGBAInto(data.data(), data.size(), out, out_size, out_stride)) {
+          throw image_load_error("webp_loader: decoding failed");
+        }
+    }
+
+// ----------------------------------------------------------------------------------------
+
+    void webp_loader::read_bgra( unsigned char *out, const size_t out_size, const int out_stride ) const
+    {
+        if (!WebPDecodeBGRAInto(data.data(), data.size(), out, out_size, out_stride)) {
```

Comment:
```suggestion
        if (!WebPDecodeBGRAInto(data.data(), data.size(), out, out_size, out_stride)) 
        {
```
Here and elsewhere, match the brace style used in the rest of dlib.